From b27a0d1b2e72e65e7b4eea1871a198c9ba3c35b8 Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Tue, 30 Jun 2020 16:13:11 +0700
Subject: [PATCH 349/351] arm64: dts: renesas: add support CSI40 IF for RZ/G2H
 and RZ/G2N

This commit is created to support CSI40 IF in HiHope RZ/G2H and
RZ/G2N platform.
This CSI is linked with OV5645 camera sensor which is controlled by
I2C channel 3.
Currently, set virtual channel 0 for ov5645 camera sensor as default.

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 .../renesas/r8a774b1-hihope-rzg2n-ex-mipi-2.1.dts  | 43 ++++++++++++++++++++++
 .../renesas/r8a774e1-hihope-rzg2h-ex-mipi-2.1.dts  | 43 ++++++++++++++++++++++
 2 files changed, 86 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a774b1-hihope-rzg2n-ex-mipi-2.1.dts b/arch/arm64/boot/dts/renesas/r8a774b1-hihope-rzg2n-ex-mipi-2.1.dts
index fcd0f75..9ab38ca 100644
--- a/arch/arm64/boot/dts/renesas/r8a774b1-hihope-rzg2n-ex-mipi-2.1.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774b1-hihope-rzg2n-ex-mipi-2.1.dts
@@ -9,6 +9,7 @@
 /dts-v1/;
 #include "r8a774b1-hihope-rzg2n-ex.dts"
 #define MIPI_PARENT_I2C_CSI20 i2c2
+#define MIPI_PARENT_I2C_CSI40 i2c3
 #include "aistarvision-mipi-adapter-2.1.dtsi"
 
 / {
@@ -77,3 +78,45 @@
 		};
 	};
 };
+
+&pfc {
+	i2c3_pins: i2c3 {
+		groups = "i2c3";
+		function = "i2c3";
+	};
+};
+
+&i2c3 {
+	pinctrl-0 = <&i2c3_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+&csi40 {
+	status = "okay";
+
+	ports {
+		port {
+			csi40_in: endpoint {
+				clock-lanes = <0>;
+				data-lanes = <1 2>;
+				remote-endpoint = <&ov5645_csi40_ep>;
+			};
+		};
+	};
+};
+
+&ov5645_csi40 {
+	enable-gpios = <&gpio1 11 GPIO_ACTIVE_HIGH>;
+	reset-gpios = <&gpio1 9 GPIO_ACTIVE_LOW>;
+
+	virtual-channel = /bits/ 8 <0>;
+
+	port {
+		ov5645_csi40_ep: endpoint {
+			clock-lanes = <0>;
+			data-lanes = <1 2>;
+			remote-endpoint = <&csi40_in>;
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/renesas/r8a774e1-hihope-rzg2h-ex-mipi-2.1.dts b/arch/arm64/boot/dts/renesas/r8a774e1-hihope-rzg2h-ex-mipi-2.1.dts
index 179b4fc..1ead5af 100644
--- a/arch/arm64/boot/dts/renesas/r8a774e1-hihope-rzg2h-ex-mipi-2.1.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774e1-hihope-rzg2h-ex-mipi-2.1.dts
@@ -9,6 +9,7 @@
 /dts-v1/;
 #include "r8a774e1-hihope-rzg2h-ex.dts"
 #define MIPI_PARENT_I2C_CSI20 i2c2
+#define MIPI_PARENT_I2C_CSI40 i2c3
 #include "aistarvision-mipi-adapter-2.1.dtsi"
 
 / {
@@ -77,3 +78,45 @@
 		};
 	};
 };
+
+&pfc {
+	i2c3_pins: i2c3 {
+		groups = "i2c3";
+		function = "i2c3";
+	};
+};
+
+&i2c3 {
+	pinctrl-0 = <&i2c3_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+&csi40 {
+	status = "okay";
+
+	ports {
+		port {
+			csi40_in: endpoint {
+				clock-lanes = <0>;
+				data-lanes = <1 2>;
+				remote-endpoint = <&ov5645_csi40_ep>;
+			};
+		};
+	};
+};
+
+&ov5645_csi40 {
+	enable-gpios = <&gpio1 11 GPIO_ACTIVE_HIGH>;
+	reset-gpios = <&gpio1 9 GPIO_ACTIVE_LOW>;
+
+	virtual-channel = /bits/ 8 <0>;
+
+	port {
+		ov5645_csi40_ep: endpoint {
+			clock-lanes = <0>;
+			data-lanes = <1 2>;
+			remote-endpoint = <&csi40_in>;
+		};
+	};
+};
-- 
2.7.4

